cpp/src/arrow/dataset/file_parquet.cc

```diff
@@ -893,20 +902,29 @@ Result<std::vector<compute::Expression>> ParquetFileFragment::TestRowGroups(
     return std::vector<compute::Expression>{};
   }
 
+  const SchemaField* schema_field = nullptr;
```

Comment:
Why move the declaration of `schema_field` outside the loop?

Reply:
Because initially I wasn't using a pointer, and that didn't work when declared inside the loop (it kept the value of the previous iteration), and then moved it out to try to fix, but the solution was to use a pointer instead, but that indeed works fine with keeping it in the loop ;) Will update.